lib/trip_dispatcher.rb

```diff
@@ -88,6 +90,57 @@ def load_trips
       end
 
       trips
+    end # end load_trips
+
+    def find_available_driver
+      @drivers.each do |driver|
+        if driver.status == :AVAILABLE
+          driver.change_status
+          return driver
+        end
```

Comment:
I'm not sure that I like this method changing the Driver's status, for two reasons.

1. The method's name is `find_available_driver`, which sounds like a query, not a command. Just based on the name I expect it to leave the data unchanged.
1. If the caller (in this case the `request_trip` method) ran into an error after this method was called but before the trip was finished, it would have to know to change the driver's status back to `:AVAILABLE`. This is a subtle dependency that appears between the two methods only in the error case. Better to let it make the change explicitly when it's ready.